[docs] Simplify state management in Text Field demo page

PratikDev opened the pull request:
Fixes #35041

[Text Field demo page](https://mui.com/material-ui/react-text-field/) had several components that weren't managing the state perfectly. some had synced values and some didn't. Since these demos' main purpose isn't to show the usage of controlled vs. uncontrolled modes, I converted all of them to uncontrolled to reduce confusion

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

michaldudak commented:
This PR makes all the components controlled. We've discussed making them uncontrolled instead to simplify the code and make the demos more focused on what their primary purpose is.

michaldudak commented:
https://reactjs.org/docs/uncontrolled-components.html

Basically, components without the `value` prop set (so they are in charge of controlling their own state changes).

PratikDev commented:
understood. as this PR has gone so blurry it could be confusing for anyone. so should I just delete this and create a new one with a more perfect description? or just commit the edits in it?

PratikDev commented:
hey @michaldudak, haven't received any feedback about this PR for a long time

michaldudak reviewed (commented):
Sorry for the delay. With so many PRs and issues, it's hard to prioritize.
I left two comments, but it looks OK in general.

PratikDev commented on docs/data/material/components/text-fields/MultilineTextFields.tsx:
```diff
@@ -24,8 +18,7 @@ export default function MultilineTextFields() {
           maxRows={4}
-          value={value}
-          onChange={handleChange}
+          defaultValue="Controlled"
```
so should remove the `defaultValue` from all the input fields?

michaldudak replied:
In cases where it's "Controlled" now, yes.

PratikDev replied:
are we again having some confusion about"controlled" and "uncontrolled" by any chance? because I think I made all of the fields "uncontrolled" as per our discussion

michaldudak replied:
Yes, that's correct; all the fields are uncontrolled. That's why having a defaultValue="Controlled" is confusing and I'm asking you to remove the `defaultValue` prop where its value is "Controlled" (literally).

PratikDev commented:
sorry for all the confusion we had. but seems we're now on the same page. please check out my recent commit. I've removed all the `defaultValue="Controlled"` and `defaultValue=""`

michaldudak commented on docs/data/material/components/text-fields/InputAdornments.js:
```diff
@@ -13,24 +13,9 @@ import Visibility from '@mui/icons-material/Visibility';
-      showPassword: !values.showPassword,
-    });
-  };
+  const handleClickShowPassword = () => setShowPassword(!showPassword);
```
Changing state based on a previous value should use the callback form:
```suggestion
  const handleClickShowPassword = () => setShowPassword((show) => !show);
```

michaldudak commented on docs/data/material/components/text-fields/InputAdornments.tsx:
```diff
@@ -59,8 +35,7 @@ export default function InputAdornments() {
             id="outlined-adornment-weight"
-            value={values.weight}
-            onChange={handleChange('weight')}
+            defaultValue=""
```
You can remove empty `defaultValue` props here as well

michaldudak commented on docs/data/material/components/text-fields/MultilineTextFields.js:
```diff
@@ -24,8 +18,7 @@ export default function MultilineTextFields() {
           maxRows={4}
-          value={value}
-          onChange={handleChange}
+          defaultValue="Controlled"
```
Remove the `defaultValue` props

PratikDev commented:
Hi, I think I removed all the `defaultValue=""` and `defaultValue="Controlled"`. and also changed the state updating in a callback form. There are no more `defaultValue=""` and `defaultValue="Controlled"` in any text fields as far I checked. but if there's anything more to change, please let me know. I'll commit them accordingly

michaldudak reviewed (approved):
Looks good to me now. Thanks for your work!

PratikDev commented:
the PR still shows Open. is there anything more to do from my end? or I'm such a noob in open source who doesn't even know how things work here?

michaldudak commented:
No, no need to do anything :) It should be included in the next release. We leave PRs open for some time after approval in case other MUI team members would like to comment.

PratikDev commented:
oh okay. got it. thanks

On Fri, Dec 9, 2022 at 7:55 PM Michał Dudak ***@***.***>
wrote:

> No, no need to do anything :) It should be included in the next release.
> We leave PRs open for some time after approval in case other MUI team
> members would like to comment.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mui/material-ui/pull/35051#issuecomment-1344337462>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUYIXMQLMV6E43CT5HTPLGTWMM25RANCNFSM6AAAAAAR2Y3GDE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
